src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -22,6 +23,10 @@ internal sealed class AdditionalSourcesCollection
 
         private static readonly StringComparer s_hintNameComparer = StringComparer.OrdinalIgnoreCase;
 
+        // Matches "/" at the beginning, relative path segments ("../", "./", "//"),
```

Comment:
Why is `/` at the beginning bad? Is it that generators shouldn't use rooted paths?

Reply:
Yes, but thinking about this again, we should probably disallow also `C:/` etc. (using a `Path` API probably).

Reply:
Correction, `C:/` is already disallowed, since colon `:` character is not allowed. Anyway, added also `Path.IsRootedPath` check.